selfie.c

```diff
@@ -5037,8 +5039,25 @@ void compile_statement() {
         type_warning(UINT64STAR_T, ltype);
 
       get_symbol();
+    // "*" "(" expression ")"
+    } else if (symbol == SYM_LPARENTHESIS) {
+      get_symbol();
+
+      ltype = compile_expression();
 
-      // "*" variable "="
+      if (ltype != UINT64STAR_T)
+        type_warning(UINT64STAR_T, ltype);
+
+      if (symbol == SYM_RPARENTHESIS)
+        get_symbol();
+      else
+        syntax_error_symbol(SYM_RPARENTHESIS);
+    } else
+      syntax_error_symbol(SYM_LPARENTHESIS);
```

Comment:
Put a `return` statement in this `else` case and then remove the subsequent check of `ltype`.